src/core/Expander/ExpanderGroup/ExpanderGroup.tsx

```diff
@@ -0,0 +1,189 @@
+import { TokensProp, InternalTokensProp } from '../../theme';
+import { HtmlDiv, HtmlButton, HtmlButtonProps, HtmlSpan } from '../../../reset';
+import { baseStyles } from './ExpanderGroup.baseStyles';
+import { VisuallyHidden } from '../../../components';
+
+const baseClassName = 'fi-expander-group';
+const openClassName = `${baseClassName}--open`;
+const expandersContainerClassName = `${baseClassName}_expanders`;
+const openAllButtonClassName = `${baseClassName}_all-button`;
+interface InternalExpanderGroupProps {
+  /** Expanders and option other components */
+  children: ReactNode;
+  /** 'Open all' button text */
+  OpenAllText: string;
+  /** 'Close all'-component (Button) */
+  CloseAllText: string;
+  /** Custom classname to extend or customize */
+  className?: string;
+  /** 'Open all' button text for screen readers, hides OpenAllText for screen readers if provided */
+  AriaOpenAllText?: string;
+  /** 'Close all' button text for screen readers, hides CloseAllText for screen readers if provided */
+  AriaCloseAllText?: string;
+  /** Open/Close all button props */
+  toggleAllButtonProps?: Omit<
+    HtmlButtonProps,
+    | 'onClick'
+    | 'onMouseDown'
+    | 'onMouseUp'
+    | 'onKeyPress'
+    | 'onKeyUp'
+    | 'onKeyDown'
+  >;
+}
+
+interface ExpanderOpenStates {
+  [key: string]: boolean;
+}
+
+type ExpanderGroupTargetOpenState = {
+  targetOpenState: boolean;
+};
+
+interface ExpanderGroupState {
+  /** Expanders that are open */
+  expanders: ExpanderOpenStates;
+  /** State change transition request */
+  expanderGroupOpenState: ExpanderGroupTargetOpenState;
+}
+
+export interface ExpanderGroupProviderState {
+  onExpanderOpenChange: (id: string, toState: boolean | undefined) => void;
+  expanderGroupOpenState: ExpanderGroupTargetOpenState;
+}
+
+const defaultProviderValue: ExpanderGroupProviderState = {
+  onExpanderOpenChange: () => null,
+  expanderGroupOpenState: {
+    targetOpenState: false,
+  },
+};
+
+const { Provider, Consumer: ExpanderGroupConsumer } = React.createContext(
+  defaultProviderValue,
+);
+
+class BaseExpanderGroup extends Component<InternalExpanderGroupProps> {
+  state: ExpanderGroupState = {
+    expanders: {},
+    expanderGroupOpenState: {
+      targetOpenState: false,
+    },
+  };
+
+  handleExpanderOpenChange = (id: string, newState: boolean | undefined) => {
+    this.setState((prevState: ExpanderGroupState) => {
+      const { expanders: prevExpanders } = prevState;
+      const expanders = Object.assign({}, prevExpanders);
+      if (newState !== undefined) {
+        expanders[id] = newState;
+        return {
+          expanders,
+        };
+      }
+      delete expanders[id];
+      return { expanders };
+    });
+  };
+
+  expandersOpenState = () => {
+    const expanderCount = Object.keys(this.state.expanders).length;
+    const openExpanderCount = Object.values(this.state.expanders).filter(
+      (value) => value === true,
+    ).length;
```

Comment:
Would it be possible to use
`(value) => !!value` or `(value) => value` instead?

Or would it cause some implications?

Reply:
Should not have any implications as values are booleans here. Changed it to the latter proposed option. It is shorter and a bit neater, but also more difficult to read as it requires one to know or check the type of value (expanders).

Reply:
Then if that is the case, should the `value` be more describing? 
Like using the same `targetOpenState` to explicitly tell what it is.

Reply:
What about `isOpen`? It would maybe better describe what the value actually is. The value has no name in the interface (`[key: string]: boolean`).

Reply:
That is better or maybe even shorter, `open.

Reply:
Changed it to `isOpen` as it suggests a boolean value.